### PR TITLE
per-user time zones

### PR DESCRIPTION
**Pull request by rjbs**
Give each user a TZ.  Try to use it when showing them a time.

### CODE REVIEW

File: synergy [hunk 1]
```diff
@@ -814,18 +814,19 @@ sub SAID_shows {
...
-  },
-);
+sub date_parser_for {
+  my ($self, $user) = @_;
+
+  my $tz = $user ? $user->time_zone : 'America/New_York';
+
+  DateTime::Format::Natural->new(
+    prefer_future => 1,
+    time_zone     => 'America/New_York',
```
Comment (wolfsage) [inline synergy]: Need to use $tz here

File: synergy [hunk 2]
```diff
@@ -876,29 +878,40 @@ sub SAID_chill {
...
   POE::Kernel->yield('save_state');
-  $self->reply("Okay, no more nagging until " . localtime $time, $arg);
+  $self->reply("Okay, no more nagging until $when", $arg);
 }
 
 sub SAID_mute {
   my ($self, $arg) = @_;
   return -1 unless $arg->{what} =~ s/\A\s*(alerts|all|\*)\s+//;
 
+  my $user = $self->resolve_name($arg->{who});
```
Comment (wolfsage) [inline synergy]: Always guaranteed to get a user back?

File: synergy [hunk 3]
```diff
@@ -936,6 +949,8 @@ sub SAID_remind {
 
   $who = $self->resolve_name($who, $arg->{who});
 
+  my $speaker = $self->resolve_name($arg->{who});
```
Comment (wolfsage) [inline synergy]: silly bug but with this usage, resolve_name will return nothing if $arg->{who} is someone named 'myself', or 'my', etc...

File: synergy (hunk 2 above)
Comment (rjbs) [inline synergy]: Yes.  It's stupid: we've already resolved the user when setting up the `$arg` that's passed in, but we pass the username, not object.  Historical reasons.  I plan to fix this.

File: synergy (hunk 3 above)
Comment (rjbs) [inline synergy]: Can't happen unless we set up such users, which I promise not to do. :-)

**Comment by rjbs** [comment]
Thanks, fixed and merged.

**Pull request merged**